indigo/indigo-extras/src/main/scala/indigoextras/ui/RadioButton.scala

```diff
@@ -0,0 +1,206 @@
+package indigoextras.ui
+
+import indigo.shared.EqualTo._
+import indigo.shared.Outcome
+import indigo.shared.datatypes.{Depth, Point, Rectangle}
+import indigo.shared.events.GlobalEvent
+import indigo.shared.input.Mouse
+import indigo.shared.scenegraph.Group
+
+/**
+  * Structure for presenting a radio button.
+  *
+  * @param buttonAssets The graphics to use identically for all of the options: up (unselected), down (selected) and over
+  * @param bounds A list of the rectangle locations of each option button by index
+  * @param depth The depth at which to present this button
+  * @param selected The index of the option currently selected, if any
+  * @param over The index of the option currently hovered over, if any
+  * @param onUnselected Events to fire when an option becomes unselected, mapping from the index of the option deselected
+  * @param onSelected Events to fire when an option becomes selected, mapping from the index of the option selected
+  * @param onHoverOver Events to fire when an option is hovered over, mapping from the index of the option hovered over
+  * @param onHoverOut Events to fire when an option is no longer hovered over, mapping from the index of the option no longer hovered over
+  */
+final case class RadioButton(
```

Comment:
I do think this should be called `RadioButtons` or `RadioButtonList` (list might be yukky...) or something. It isn't a single radio button.

Reply:
OK. Would RadioButtonGroup (evoking the terminology of old Java Swing...) be OK?

Reply:
It's as good as anything!

Reply:
Done!